engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -242,6 +244,10 @@ public function __construct(\Elgg\AutoloadManager $autoload_manager) {
 		});
 
 		$this->setClassName('widgets', '\Elgg\WidgetsService');
+		
+		$this->setFactory('uploads', function(ServiceProvider $c) {
+			return $c->request->files;
```

Comment:
Oh I thought someone specifically requested only to expose files. I'm all for making more services public.